README.md

```diff
@@ -1,15 +1,126 @@
+
+What this did is generate a new folder under `zomes` called `users`. Here is the folder structure of it.
+- users
+  - code
+    - src
+      - main.rs
+    - .build
+    - Cargo.toml
+  - zome.json
+
+So in a given Zome we have two things:
+1. a JSON file, `zome.json`, which defines and configures the Zome
+2. a `code` folder, which can be compiled into a single `WASM` file with the code for this Zome
+
+In order for Holochain to run your app, you have to build your code into a single packaged file. Those instructions follow.
+
+## What are .hcpkg files?
+
+A Holochain app can be fully contained in a file known as a `.hcpkg` file.
+It is a JSON file, with a particular structure that Holochain can understand, and execute.
+
+This is an unusual JSON file, it is part configuration, and part executable.
+
+The configuration part comes from the `json` files that are throughout your app. One at the top level for the application (`app.json`) and one for each Zome. Ultimately, these get stitched together into a single tree structure in the `.hcpkg` file.
+
+The executable part comes from having embedded Base64 encoded WebAssembly code in the file. *What does that mean?* [WebAssembly](https://webassembly.org/) is a fast and secure low-level language.
+Rather than storing the code in its raw format, Holochain expects the code to be [encoded using Base64](https://en.wikipedia.org/wiki/Base64) , so that its as compact as possible.
+
+If you haven't heard of WebAssembly (WASM for short), that's ok. Important to know is that WASM is intended as a "compilation target" for other languages, not a language to write code in. So instead of writing code in WASM, write code in a language that's familiar to you, and [supports WASM](https://github.com/appcypher/awesome-wasm-langs). When it's time to run your code in Holochain, compile it.
+
+In order to avoid having to handcraft this complex JSON structure, with lots of room for error, the `hcdev package` command streamlines the process of taking your "raw" application folder, and packaging it up into the final `.hcpkg` file. 
+
+More information about this follows.
+
+## Using Built-in Compilation
+
+The `hcdev package` tool will automate the process of compiling your Zome code, encoding it, and inserting into the `.hcpkg` file. In order to get these benefits, you just need to make sure that you have the right compilation tools installed on the machine you are using the command line tools from, and that you have the proper configuration files in your Zome folders. 
+
+`hcdev package` works with special files called `.build` files. 
+
+### .build files
+In the process of building a `.hcpkg` file, here is what Holochain does.
+- It iterates Zome by Zome adding them to the JSON
+- For each Zome, it looks for a `code` folder
+- Within that code folder, it looks for a `.build` file
```

Comment:
It actually would replace any folder's contents with a build artifact if that folder contains a `.build` file - not only `code` folders in zomes...

Reply:
Right... it wouldn't have to be named `code` so I could generalize that

Reply:
Although it is IMPORTANT that it's called `code` (the folder) I think because it uses the folder name as the key for its value